### PR TITLE
Fix mixing segwit support for ledger

### PR DESCRIPTION
**Pull request by romaniy173**
Issue: add support to signer to allow mixing native and nested segwit for ledger tx

### CODE REVIEW

File: BlocksettleNetworkingLib/HeadlessContainer.cpp [hunk 1]
```diff
@@ -458,8 +458,8 @@ bs::signer::RequestId HeadlessContainer::signMultiTXRequest(const bs::core::wall
 
    headless::SignTXMultiRequest request;
    for (const auto &input : txMultiReq.inputs) {
-      request.add_walletids(input.second);
-      signer.addSpender(std::make_shared<ScriptSpender>(input.first));
+      request.add_walletids(input.walletId_);
+      signer.addSpender(std::make_shared<ScriptSpender>(input.utxo_));
```
Comment (sergey-chernikov) [inline BlocksettleNetworkingLib/HeadlessContainer.cpp:462]: We don't expose publicly accessible struct members with "_" suffix. So better to remove it.

File: WalletsLib/CoreWallet.cpp [hunk 2]
```diff
@@ -1190,6 +1190,70 @@ BinaryData bs::core::SignMultiInputTX(const bs::core::wallet::TXMultiSignRequest
...
+
+   for (const auto &recipient : txMultiReq.recipients) {
+      signer.addRecipient(recipient);
+   }
+
+   signer.sign();
+
+   for (const auto &spender : spenders) {
+      const auto &itSig = inputSigs.find(spender.first);
+      if (itSig == inputSigs.end() || !spender.second->isSegWit()) {
```
Comment (sergey-chernikov) [inline WalletsLib/CoreWallet.cpp:1245]: Does this function work for SegWit inputs only? You have code above for legacy spenders init.

Comment (romaniy173) [reply WalletsLib/CoreWallet.cpp:1245]: Yeah, if we are talking about mixing(more then one input type) - we use only this for segwit, so no segwit and legacy input mixing for now. If function about, it's about signing from one wallet - that function is possible for legacy

Comment (romaniy173) [reply WalletsLib/CoreWallet.cpp:1245]: True, I will delete those function, this should be possible only for segwit

Comment (romaniy173) [reply WalletsLib/CoreWallet.cpp:1245]: Done

File: BlocksettleNetworkingLib/HeadlessContainer.cpp (hunk 1 above)
Comment (romaniy173) [reply BlocksettleNetworkingLib/HeadlessContainer.cpp]: Done

**Pull request merged**